tests/infra/testmode.py

```diff
@@ -0,0 +1,7 @@
+import os
+import unittest
+
+
+def controlled_access(f):
```

Comment:
Seems like a good idea and I'll likely add something like this as well, until I'm guessing our code bases share enough that it makes sense to merge.